ComponentKit/Utilities/CKComponentActionInternal.h

```diff
@@ -21,50 +21,74 @@
  We support several different types of action variants. You don't need to use this value anywhere, it's set for you
  by whatever initializer you end up using.
  */
-typedef NS_ENUM(NSUInteger, _CKTypedComponentActionVariant) {
-  _CKTypedComponentActionVariantRawSelector = 0,
-  _CKTypedComponentActionVariantTargetSelector,
-  _CKTypedComponentActionVariantComponentScope
+typedef NS_ENUM(NSUInteger, CKTypedComponentActionVariant) {
+  CKTypedComponentActionVariantRawSelector = 0,
+  CKTypedComponentActionVariantTargetSelector,
+  CKTypedComponentActionVariantComponentScope
+};
+
+typedef NS_ENUM(NSUInteger, CKComponentActionSendBehavior) {
+  /** Starts searching at the sender's next responder. Usually this is what you want to prevent infinite loops. */
+  CKComponentActionSendBehaviorStartAtSenderNextResponder,
+  /** If the sender itself responds to the action, invoke the action on the sender. */
+  CKComponentActionSendBehaviorStartAtSender,
+};
+
+struct CKTypedComponentActionValue {
+  CKTypedComponentActionValue() : _variant(CKTypedComponentActionVariantRawSelector), _target(nil), _scopeHandle(nil), _selector(NULL) {}
```

Comment:
I might consider outlining this if it's going to show up in lots of end-user code; it's at least 4 instructions, whereas the function call should take 2. Can always wait on that though.